library/Icingadb/Widget/Detail/ObjectDetail.php

```diff
@@ -602,4 +605,55 @@ protected function fetchCustomVars()
             $this->object->customvar_flat = $customvarFlat->execute();
         }
     }
+
+    /**
+     * Create a list of root problems of the object that is unreachable because of dependency failure
+     *
+     * @return ?BaseHtmlElement[]
+     */
+    protected function createRootProblems(): ?array
+    {
+        // If a dependency has failed, then the children are not reachable. Hence, the root problems should not be shown
+        // if the object is reachable. And in case of a service, since, it may be also be unreachable because of its
+        // host being down, only show its root problems if it's really caused by a dependency failure.
+        if (
+            $this->object->state->is_reachable
+            || ($this->object instanceof Service && ! $this->object->has_problematic_parent)
+        ) {
+            return null;
+        }
+
+        $rootProblems = UnreachableParent::on($this->getDb(), $this->object)
+            ->with([
+                'redundancy_group',
+                'redundancy_group.state',
+                'host',
+                'host.state',
+                'host.icon_image',
+                'host.state.last_comment',
+                'service',
+                'service.state',
+                'service.icon_image',
+                'service.state.last_comment',
+                'service.host',
+                'service.host.state'
+            ])
+            ->setResultSetClass(VolatileStateResults::class)
+            ->orderBy([
+                'host.state.severity',
+                'host.state.last_state_change',
+                'service.state.severity',
+                'service.state.last_state_change',
+                'redundancy_group.state.last_state_change'
```

Comment:
one small thing: just ordering after a group's state change date doesn't suffice. A group doesn't have a severity, but still a state, which should be used instead.